lib/components/viewers/nearby/nearby-view.tsx

```diff
@@ -0,0 +1,177 @@
+import { connect } from 'react-redux'
+import { FormattedMessage, IntlShape, useIntl } from 'react-intl'
+import { Location } from '@opentripplanner/types'
+import { MapRef, useMap } from 'react-map-gl'
+import React, { useEffect, useRef, useState } from 'react'
+
+import * as apiActions from '../../../actions/api'
+import * as locationActions from '../../../actions/location'
+import * as uiActions from '../../../actions/ui'
+import { AppReduxState } from '../../../util/state-types'
+import Loading from '../../narrative/loading'
+import MobileContainer from '../../mobile/container'
+import MobileNavigationBar from '../../mobile/navigation-bar'
+
+import {
+  FloatingLoadingIndicator,
+  NearbySidebarContainer,
+  Scrollable
+} from './styled'
+import RentalStation from './rental-station'
+import Stop from './stop'
+import Vehicle from './vehicle-rent'
+import VehicleParking from './vehicle-parking'
+
+const AUTO_REFRESH_INTERVAL = 15000
+
+type LatLonObj = { lat: number; lon: number }
+
+type Props = {
+  fetchNearby: (latLon: LatLonObj, map?: MapRef) => void
+  getCurrentPosition?: (
+    intl: IntlShape,
+    setAsType?: string | null,
+    onSuccess?: (position: GeolocationPosition) => void
+  ) => void
+  hideBackButton?: boolean
+  mobile?: boolean
+  nearby: any
+  nearbyViewCoords?: LatLonObj
+  setHighlightedLocation: (location: Location | null) => void
+  setMainPanelContent: (content: number) => void
+  viewNearby?: (pos: LatLonObj) => void
+}
+
+const getNearbyItem = (place: any) => {
+  switch (place.__typename) {
+    case 'RentalVehicle':
+      return <Vehicle key={place.id} vehicle={place} />
+    case 'Stop':
+      return <Stop showOperatorLogo stopData={place} />
+    case 'VehicleParking':
+      return <VehicleParking place={place} />
+    case 'BikeRentalStation':
+      return <RentalStation place={place} />
+    default:
+      console.warn(
+        `Received unsupported nearby place type: ${place.__typename} `
+      )
+      return null
+  }
+}
+
+function NearbyView(props: Props): JSX.Element {
+  const {
+    fetchNearby,
+    getCurrentPosition: getPosition,
+    mobile,
+    nearby,
+    nearbyViewCoords,
+    setHighlightedLocation,
+    setMainPanelContent,
+    viewNearby
+  } = props
+  const map = useMap().current
+  const intl = useIntl()
+  const [loading, setLoading] = useState(true)
+  const firstItemRef = useRef<HTMLDivElement>(null)
+  useEffect(() => {
+    firstItemRef.current?.scrollIntoView({ behavior: 'smooth' })
+    if (nearbyViewCoords) {
+      fetchNearby(nearbyViewCoords, map)
+      setLoading(true)
+      const interval = setInterval(() => {
+        fetchNearby(nearbyViewCoords, map)
+        setLoading(true)
+      }, AUTO_REFRESH_INTERVAL)
+      return function cleanup() {
+        clearInterval(interval)
+      }
+    } else {
+      if (getPosition && viewNearby) {
+        getPosition(intl, null, (pos) => {
```

Comment:
I made some improvements here but I think it's not perfect yet. The location now falls back to the map center, but I think it needs a few more things to be good:
- Should show a marker at the current position. What symbol should we use?
- When you move the map, it should offer to update the location. 
- When you leave Nearby and come back to it, it currently shows the stale data since it's saved in Redux. The redux data should be cleared or otherwise hidden when you reopen the page.